### PR TITLE
[WIP] adding support for MoE models

### PR DESCRIPTION
**Pull request by d0rbu**
this is my current branch which adds support for MoE models and accessing their routers. later i will prepare this for a proper PR, but in the meantime i will make sure to keep it updated with main regularly.

### CODE REVIEW

**Comment by d0rbu** [comment]
yeah thats very reasonable! though to be transparent it's not my top priority atm so can't guarantee i can get to it in a timely manner (as has already been the case 😅)

**Comment by Butanium** [comment]
No worries! Thanks for your contribution!
And tbc no need to have both logits and probs in the same PR
